draft/jumpstarter-prev-draft.rst

```diff
@@ -0,0 +1,219 @@
+
+Before work is to begin on this next generation of Celery, it's crucial to have
+a discussion about and carve out the building blocks that will form the
+primitives for the upcoming Celery. These building blocks should fulfill 1, 2,
+and 3 (especially) above, and also help fulfill the goals from the linked
+`Next-Gen Rationale`_.
+
+For Celery to move forward into the modern age, and the future of computing
+with increasingly decentralized systems, it's important for Celery to follow a
+standardized asynchronous execution model. Namely, we want to define things
+like: 
+
+1. How does Celery interact with itself (its own internal components)? 
+
+2. How does Celery interact with others (and how do others interact with it)? 
+
+3. How do Celery's internal components interact with each other and their outside environment? 
+
+Also, we want Celery to work with the emerging and maturing asynchronous python
+landscape.  Celery is *very asynchronous* by nature, dealing with brokers,
+queues, results, timeouts, chord-like joins, task groups, you name it.
+Modern asynchronous python frameworks have provided new, innovative, and robust
+solutions to many things that overlap with Celery's goals.
+
+To do all of this, we propose modeling next-gen Celery off of the `Actor Model`_. Namely,
+we propose that the Celery Worker will be modeled as an `Actor System`_ of sorts.
+
+By the end of this initial implementation, we should have a proposal and reference implementation that: 
+
+1. Models Tasks and other key Celery primitives (some to be implemented and
+further specced-out down the line) as Actors. 
+
+2. Can give a clear overview about how these various Celery primitives (as
+Actors) communicate with each other and pass messages (referencing the Actor
+model). 
+
+3. Can enable the development of a future version of Celery where it can be run
+embedded with ``async/await`` (if desired by the programmer, not necessary),
+and can have synchronous-friendly workers and asynchronous-friendsly/purely
+asynchronous workers. 
+
+
+Specification
+=============
+
+TODO
+
+This section should contain a complete, detailed technical specification should
+describe the syntax and semantics of any new feature.  The specification should
+be detailed enough to allow implementation -- that is, developers other than the
+author should (given the right experience) be able to independently implement
+the feature, given only the CEP.
+
+Motivation
+==========
+
+TODO (WIP)
+
+This section should explain *why* this CEP is needed. The motivation is critical
+for CEPs that want to add substantial new features or materially refactor
+existing ones.  It should clearly explain why the existing solutions are
+inadequate to address the problem that the CEP solves.  CEP submissions without
+sufficient motivation may be rejected outright.
+
+Let's start with an example, and then generalize. A quick search in Celery's
```

Comment:
@thedrow I'd like to potentially move this section back into the CEP probably in the next revision. However, I wanted to ask if you think this is a good idea or not. Is it too out of scope? I really like a bit of the discussion on Celery's internals and issues that could be resolved by more SRP code, and why something new is needed, but maybe it's out of scope and should be reserved for a future CEP?

Reply:
Also, if I did, I would rephrase things in more formal language in certain sections per your comment.